questModal MVP

zfreeman341 opened the pull request:
-Display current Quest
- Select Quest
- Display current Objective 
- Increment and decrement current objective

BenTracyDotCom commented on src/screens/Campaign.jsx:
```diff
@@ -1,27 +1,62 @@
       <Header />
+      <Text style={styles.currentQuestTitle}>Current Quest: {currentQuest}</Text>
+      <Text>Current Objective: {currentQuestData && currentObjectiveIndex !== undefined ? currentQuestData.objectives[currentObjectiveIndex] : 'None'}</Text>
+      <Button title="Next Objective" onPress={handleNextObjective}></Button>
```
Buttons aren't as nice and responsive as a TouchableOpacity and I'm hoping to reuse all of our "button" styling throughout for consistency. Can you use a: 
<TouchableOpacity>
  <Text>Next Objective</Text>
</TouchableOpacity> 
format instead?

zfreeman341 replied:
Updated

BenTracyDotCom commented on utilities/sampleData/sampleCampaign.js:
```diff
@@ -36,15 +36,31 @@ export default {
+      objectives: ['Escort Cart to Phandalin', 'Run to Phandolin', 'Defeat Redbrands', 'Seduce the barmaid']
+    },
+    {
+      title: 'The Lost Mine of Wave Echo',
```
This looks great!

BenTracyDotCom commented on src/screens/Campaign.jsx:
```diff
@@ -1,27 +1,62 @@
+      <Text>Current Objective: {currentQuestData && currentObjectiveIndex !== undefined ? currentQuestData.objectives[currentObjectiveIndex] : 'None'}</Text>
+      <Button title="Next Objective" onPress={handleNextObjective}></Button>
+      <Button title="Previous Objective" onPress={handlePreviousObjective}></Button>
+      <Button title="Show Quests" onPress={toggleQuestModal}></Button>
```
Add an onPress handler to the Quest TouchableOpacity to either toggle this modal or navigate to a separate screen with buttons pertinent to quest management instead of creating a new button

zfreeman341 replied:
I added a TouchableOpacity modal for now, long-term we can add a journal page instead of (or in addition to) a quest modal for quest management

BenTracyDotCom commented on src/features/campaigns/campaignSlice.js:
```diff
@@ -78,10 +79,33 @@ const campaignSlice = createSlice({
-export const { currentQuestUpdated, addNote, deleteNote, editNote, addCharNote, editCharNote, setActiveNotes, setCurrentCampaign, addNPC, editNPC, deleteNPC } = campaignSlice.actions
+export const { currentQuestUpdated, addNote, deleteNote, editNote, addCharNote, editCharNote, setActiveNotes, setCurrentCampaign, addNPC, editNPC, deleteNPC, setCurrentObjectiveIndex, nextObjective, previousObjective } = campaignSlice.actions
 
 export default campaignSlice.reducer
```
Add a state value and action to manage questModalVisible, then if we decide other screens should be able to manage the modal it'll be easy to integrate

BenTracyDotCom commented on src/features/quests/QuestModal.js:
```diff
@@ -0,0 +1,83 @@
+          <FlatList
+            data={quests}
+            renderItem={({ item, index: questIndex }) => (
+              <TouchableOpacity onPress={() => handleQuestPressLocal(item)}>
```
This FlatList functionality is awesome!

BenTracyDotCom commented on src/features/quests/QuestModal.js:
```diff
@@ -0,0 +1,83 @@
+                  <FlatList
+                    data={item.objectives}
+                    renderItem={({ item: objective, index: objectiveIndex }) => (
+                      <TouchableOpacity onPress={() => handleObjectivePress(questIndex, objectiveIndex)}>
```
this is throwing an "undefined is not a function" error

BenTracyDotCom reviewed (changes requested):
-Campaign screen appearance shouldn't change, add quest modification functionality to either a modal or a separate screen
-Love the collapsing quest/objectives menu, just need a '+' at the end of each list to add a quest/objective

zfreeman341 commented:
> 

I think the campaign screen appearance should update to reflect the currentQuest and currentObjective. That's information that I think is good for a DM to keep front-of-mind. It's probably taking up a little too much real estate right now, but that's something we can change later.

The addQuest and addObjective functionality is taking a bit longer than expected. I wanted to get a PR out before the weekend, but I'll add that functionality next week.

BenTracyDotCom commented:
The campaign screen already reflects currentQuest and currentObjective, all that info is in the top maybe 1/3 of the screen (by 'Quest:' and 'Objective:'). Campaign and character notes should take up the majority of the campaign overview screen

zfreeman341 commented:
I've removed the new text from the "Campaign" compnent, and added the objective changing logic to the Header component. Will add the "add quest" and "add objective" logic later this week, but as is we should be good to merge in

BenTracyDotCom commented:
handleQuest and handleObjective in campaigns/Header.jsx should do what the "show quests" button does, i.e. just pop up the quests modal (which looks great). We don't need "previous quest" and "next quest" buttons because that modal should have all the info i need: either I know where my party is going next, or (as is more often the case), I have three or four "objectives" that the party can choose between or do in any order to further the bigger quest. The way to change the active objective should be to either click one of the sub-objectives in the quests modal or to add a new objective to the current quest through that same modal

BenTracyDotCom commented:
Looks good, thanks!
